src/main/scala/org/neo4j/spark/service/MappingService.scala

```diff
@@ -68,19 +68,19 @@ class Neo4jWriteMappingStrategy(private val options: Neo4jOptions)
   private def keysStrategyConsumer(): MappingBiConsumer = new MappingBiConsumer {
     override def accept(key: String, value: AnyRef): Unit = {
       if (options.relationshipMetadata.source.nodeKeys.contains(key)) {
-        sourceNodeMap.get(KEYS).put(key, value)
+        sourceNodeMap.get(KEYS).put(options.relationshipMetadata.source.nodeKeys.getOrElse(key, key), value)
       }
       else if (options.relationshipMetadata.source.nodeProps.contains(key)) {
-        sourceNodeMap.get(PROPERTIES).put(key, value)
+        sourceNodeMap.get(PROPERTIES).put(options.relationshipMetadata.source.nodeProps.getOrElse(key, key), value)
       }
```

Comment:
can we move this to a function like this?

```scala
private def addToNodeMap(nodeMap: Map[String, AnyRef], nodeMetadata: Neo4jNodeMetadata, key: String, value: AnyRef): Unit = {
      if (nodeMetadata.nodeKeys.contains(key)) {
        nodeMap.get(KEYS).put(nodeMetadata.nodeKeys.getOrElse(key, key), value)
      }
      if (nodeMetadata.nodeProps.contains(key)) {
        nodeMap.get(PROPERTIES).put(nodeMetadata.nodeProps.getOrElse(key, key), value)
      }
}
```

plus removing the else (also at line 82) we will allow to put the same properties in source and/or target and/or relationship, I think it would be useful.

Reply:
Ok for removing the else.

I don't extracting the function useful in this case, it will just move the existing code somewhere else, won't add any performance or readability improves. I think the code here is pretty straightforward. We can extract long lines into variables to make it more easy to read. Wdyt?